src/data.rs

```diff
@@ -0,0 +1,30 @@
+use data_encoding::HEXUPPER;
+use sha1::{Digest, Sha1};
+use std::{fs, io, path};
+
+pub const GIT_DIR: &str = ".rustig";
+
+pub fn init() -> io::Result<()> {
+    fs::create_dir(GIT_DIR)?;
+    let object_path = path::PathBuf::from(format!("{GIT_DIR}/objects"));
+    fs::create_dir(object_path)?;
+    Ok(())
+}
+
+pub fn hash_object(path: path::PathBuf) -> io::Result<String> {
+    let content = fs::read_to_string(path).expect("Unable to read file");
+
+    let hash = Sha1::new().chain_update(&content).finalize();
+    let oid = HEXUPPER.encode(hash.as_ref());
+
+    let output_path = path::PathBuf::from(format!("{GIT_DIR}/objects/{oid}"));
+    fs::write(output_path, content).expect("Unable to write file");
+
+    Ok(oid)
+}
+
+pub fn get_object(object: String) -> io::Result<String> {
+    let object_path = path::PathBuf::from(format!("{GIT_DIR}/objects/{object}"));
+    let content = fs::read_to_string(object_path).expect("Unable to read file");
+    Ok(content)
```

Comment:
```suggestion
    return Ok(content);
```

> Last expression returns
> You can return early from a function by using the return keyword and specifying a value, but most functions return the last expression implicitly.

Je trouve cette écriture peu explicite, WDTY ?

Reply:
En Rust c'est idiomatique d'utiliser la dernière ligne de la fonction pour retourner une valeur. Donc on omet le mot clé return et le `;`
Sauf dans les cas de early return.